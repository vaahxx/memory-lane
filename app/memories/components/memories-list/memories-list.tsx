import { MemoryCard } from "../memory-card/memory-card";
import { Memory } from "../../../models/memory.model";

type MemoriesListProps = {
  memories: Memory[];
};

export function MemoriesList({ memories }: MemoriesListProps) {
  if (!memories?.length) {
    return (
      <>
        <span>You have no memories yet.</span>
        <span>Create one!</span>
      </>
    );
  }

  return (
    <>
      {memories.map((memory: Memory) => (
        <MemoryCard
          key={memory.id}
          title={memory.title}
          content={memory.description}
          date={memory.timestamp}
          image={memory.image_url}
        />
      ))}
    </>
  );
}
