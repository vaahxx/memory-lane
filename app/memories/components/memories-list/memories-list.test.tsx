import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoriesList } from "./memories-list";

const mockMemories = [
  {
    id: 1,
    title: "Memory 1",
    description: "Description 1",
    timestamp: "2023-01-01T12:00:00Z",
    image_url: "/images/memory1.jpg",
    user_id: 1,
  },
  {
    id: 2,
    title: "Memory 2",
    description: "Description 2",
    timestamp: "2023-01-02T12:00:00Z",
    image_url: "/images/memory1.jpg",
    user_id: 1,
  },
];

describe("MemoriesList Component", () => {
  it("renders a message when there are no memories", () => {
    render(<MemoriesList memories={[]} />);

    expect(screen.getByText("You have no memories yet.")).toBeInTheDocument();
    expect(screen.getByText("Create one!")).toBeInTheDocument();
  });

  it("renders memory cards when memories are provided", () => {
    render(<MemoriesList memories={mockMemories} />);

    expect(screen.getByText("Memory 1")).toBeInTheDocument();
    expect(screen.getByText("Memory 2")).toBeInTheDocument();

    expect(screen.getByText("Description 1")).toBeInTheDocument();
    expect(screen.getByText("Description 2")).toBeInTheDocument();
    expect(screen.getByText("January 1, 2023")).toBeInTheDocument();
    expect(screen.getAllByRole("img")[0]).toMatchSnapshot();
  });
});
