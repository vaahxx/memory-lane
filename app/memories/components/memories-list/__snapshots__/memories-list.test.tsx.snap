// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoriesList Component renders memory cards when memories are provided 1`] = `
<img
  alt="Memory"
  class="object-cover rounded-full shadow border-2 w-[100px] h-[100px]"
  data-nimg="1"
  decoding="async"
  height="100"
  loading="lazy"
  src="/_next/image?url=%2Fimages%2Fmemory1.jpg&w=256&q=75"
  srcset="/_next/image?url=%2Fimages%2Fmemory1.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fmemory1.jpg&w=256&q=75 2x"
  style="color: transparent;"
  width="100"
/>
`;
